sql/src/main/java/com/vmware/ddlog/DDlogJooqProvider.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2018-2020 VMware, Inc. All Rights Reserved.
+ * SPDX-License-Identifier: BSD-2
+ */
+package com.vmware.ddlog;
+
+import ddlogapi.DDlogAPI;
+import ddlogapi.DDlogException;
+import ddlogapi.DDlogRecord;
+import org.jooq.DSLContext;
+import org.jooq.Field;
+import org.jooq.Record;
+import org.jooq.Result;
+import org.jooq.Table;
+import org.jooq.impl.DSL;
+import org.jooq.tools.jdbc.MockDataProvider;
+import org.jooq.tools.jdbc.MockExecuteContext;
+import org.jooq.tools.jdbc.MockResult;
+
+import javax.annotation.Nullable;
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class DDlogJooqProvider implements MockDataProvider {
+    private static final String INTEGER_TYPE = "java.lang.Integer";
+    private static final String STRING_TYPE = "java.lang.String";
+    private static final String BOOLEAN_TYPE = "java.lang.Boolean";
+    private static final String LONG_TYPE = "java.lang.Long";
+
+    private final DDlogAPI dDlogAPI;
+    private final DSLContext dslContext;
+    private final Map<String, List<Field<?>>> tables = new HashMap<>();
+
+    public DDlogJooqProvider(final DDlogAPI dDlogAPI, final List<String> sqlStatements) {
```

Comment:
I don't know this Jooq stuff, but I would expect there's a way to mock just input tables, not statements, and then statements can run on these tables.

Reply:
This affects the API a bit. Here, I the user supply the SQL statements to make a schema out of (and we extract the information we need it from it, like the mapping of table names to JOOQ Fields). The alternative is that the user supplies a JOOQ connection with the schema already initialized.